File: backend-py/api/business/category.py
from fastapi import APIRouter, HTTPException
from database import get_conn

router = APIRouter(prefix="/api/business/category", tags=["Business Category"])

@router.get("/getAll")
def get_categories():
    try:
        with get_conn() as conn:
            with conn.cursor() as cur:
                cur.execute("SELECT category_id, category_name FROM business_categories")
                categories = cur.fetchall()

        return {
            "success": True,
            "categories": categories
        }

    except Exception as e:
        raise HTTPException(status_code=500, detail=f"เกิดข้อผิดพลาด: {str(e)}")

File: backend/api/business/category.js
const express = require('express');
const router = express.Router();
const { pool } = require('../../database/database');

router.get('/getAll', async (req, res) => {
  try {
    const [categories] = await pool.query(
      'SELECT category_id, category_name FROM business_categories'
    );
    res.json({
      success: true,
      categories
    });
  } catch (e) {
    res.status(500).json({
      success: false,
      detail: `เกิดข้อผิดพลาด: ${e.message}`
    });
  }
});

module.exports = router;
